api-guidelines/deprecation.md

```diff
@@ -0,0 +1,65 @@
+Sometimes it is necessary to phase out an API endpoint (or version), for
+instance, if a field is no longer supported in the result or a whole
+business functionality behind an endpoint has to be shut down. There are
+many other reasons as well. As long as these endpoints are still used by
+consumers these are breaking changes and not allowed. Deprecation rules
+have to be applied to make sure that necessary consumer changes are
+aligned and deprecated endpoints are not used before API changes are
+deployed.
+
+# MUST Obtain Approval of Clients
+
+Before shutting down an API (or version of an API) the producer must
+make sure that all clients have given their consent to shut down the
+endpoint. Producers should help consumers migrate to a new
+endpoint (i.e. by providing a migration manual). After all clients are
+migrated, the producer may shut down the deprecated API.
+
+# MUST Agree Deprecation Timespan with External Partners
```

Comment:
Do we give a deprecation timespan if we have to obtain approval from clients before deprecating?

Reply:
this probably differs on a case by case basis

Reply:
yes, this is for discussion with the consumers so not specifying in these docs.